web/src/echo.js

```diff
@@ -217,14 +218,6 @@ export function try_deliver_locally(message_request, insert_new_messages) {
     // echoed; in that case, the compose box will remain in the
     // sending state rather than being cleared to allow composing a
     // next message.
-    //
-    // Notably, this algorithm will allow locally echoing a message in
-    // cases where we are currently looking at a search view where
-    // `!filter.can_apply_locally(message)`; so it is possible for a
-    // message to be locally echoed but not appear in the current
-    // view; this is useful to ensure it will be visible in other
-    // views that we might navigate to before we get a response from
-    // the server.
```

Comment:
Hmm, looking at this comment, I think this change will have a bad interaction with the "jump to sent message" behavior -- we do want your newly sent message to be immediately visible in that new view.

Reply:
This only affects the "immediately visible" behavior in the case someone sends a message when narrowed to search view. 

And that part was not working at all, actually earlier we were jumping to sent message view even before message locally echoed (due to 400 error). So, this is improvement in every sense..

Let me know if I'm missing something.

Reply:
We should local echo but avoid that server request if the message id is a local echoed one. Also, it makes sense to do the later regardless of if we local echo or not here.

Reply:
See the [updated approach](https://github.com/zulip/zulip/pull/30813#issuecomment-2233149483).